vmm/src/cpu.rs

```diff
@@ -676,9 +681,9 @@ impl CpuManager {
 
         // Adding vCPU to the CpuManager's vCPU list.
         let vcpu = Arc::new(Mutex::new(vcpu));
-        self.vcpus.push(Arc::clone(&vcpu));
+        self.vcpus.push(vcpu);
 
-        Ok(vcpu)
```

Comment:
It looks like a separate refactoring, and should be put in a separate commit?  

I was confused it (`vmm::cpu::CpuManager::create_vcpu()`) with `hypervisor::vm::Vm::create_vcpu()`  while I was reading the patch at the beginning..